sklearn/ensemble/forest.py

```diff
@@ -402,12 +418,66 @@ def __init__(self,
             estimator_params=estimator_params,
             bootstrap=bootstrap,
             oob_score=oob_score,
+            permutation_feature_importances=permutation_feature_importances,
             n_jobs=n_jobs,
             random_state=random_state,
             verbose=verbose,
             warm_start=warm_start,
             class_weight=class_weight)
 
+    def _set_permutation_feature_importances(self, X, y):
+        # Convert data to arrays, y to row vector
+        X = np.array(X)
```

Comment:
X will either be an array or sparse matrix already. If sparse matrix, we should be raising a `NotImplementedError`, I think